test/transformer/exclude.js

```diff
@@ -0,0 +1,82 @@
+var Lab = require('lab');
+var lab = exports.lab = Lab.script();
+var describe = lab.describe;
+var it = lab.it;
+var beforeEach = lab.beforeEach;
+var afterEach = lab.afterEach;
+var expect = require('code').expect;
+var sinon = require('sinon');
+
+var Transformer = require('../../lib/transformer');
+
+describe('Transformer', function() {
+  describe('exclude', function() {
+    var transformer;
+
+    beforeEach(function (done) {
+      transformer = new Transformer('/etc', []);
+      sinon.stub(transformer, 'addWarning');
```

Comment:
philosophical question: when unit testing, since all these things are within one object (Transformer), why completely stub out `addWarning`? I definitely think you should be _spying_ on it, but I don't think you necessarily need to stub it out completely. (as an aside, if it was say logging things to some service, I might think differently: would you _stub_ that service module to make sure it was called, while allowing `addWarning` to complete? this is thinking kinda in the way jest works, where everything in the file you wrote is available to you, and everything else is stubbed. Since `addWarning` is _in_ `Transformer`, why not let it do it's thing? could be that what it returns/does effects the function you wrote in a subtle way as well.)